ste/xfer-deleteBlobFS.go

```diff
@@ -51,7 +54,23 @@ func doDeleteHNSResource(jptm IJobPartTransferMgr) {
 	transferDone := func(err error) {
 		status := common.ETransferStatus.Success()
 		if err != nil {
-			status = common.ETransferStatus.Failed()
+			var respErr *azcore.ResponseError
+			if errors.As(err, &respErr) {
+				if respErr.StatusCode == http.StatusNotFound {
+					// if the delete failed with err 404, i.e resource not found, then mark the transfer as success.
+					status = common.ETransferStatus.Success()
+				}
+				// If the status code was 403, it means there was an authentication error and we exit.
+				// User can resume the job if completely ordered with a new sas.
+				if respErr.StatusCode == http.StatusForbidden {
+					errMsg := fmt.Sprintf("Authentication Failed. The SAS is not correct or expired or does not have the correct permission %s", err.Error())
```

Comment:
Should we assume its only SAS, what about expired OAuth tokens here. Error message shall be generic if any other auth mode also fails with similar error here.

Reply:
In numerous instances, particularly when addressing the deletion of a blob or file, our error handling strategy tends to focus on handling 404 and 403 errors exclusively. Currently, the error message remains intentionally generic to accommodate various scenarios apart from failed snapshots in some instances.